src/plugins/auto/src/auto_schedule.cpp

```diff
@@ -139,10 +143,55 @@ void AutoSchedule::init() {
             // if have CPU Device,  enable m_compile_context[CPU]
             if (cpu_iter != m_context->m_device_priorities.end()) {
                 m_compile_context[CPU].m_is_enabled = true;
-                m_compile_context[CPU].m_device_info = *cpu_iter;
-                m_compile_context[CPU].m_device_info.config[ov::hint::performance_mode.name()] = ov::hint::PerformanceMode::LATENCY;
-                m_compile_context[CPU].m_worker_name = "CPU_HELP";
-                LOG_INFO_TAG("will load CPU for accelerator");
+                if (!is_actual_cpu) {
+                    // user does not set the compiling threads
+                    // limit the threads num for compiling
+                    auto device = m_compile_context[ACTUALDEVICE].m_device_info.device_name;
+                    auto& device_config = m_compile_context[ACTUALDEVICE].m_device_info.config;
+                    if (is_actual_gpu) {
+                        int max_threads = 0;
+                        try {
+                            max_threads = m_context->m_ov_core->get_property(device, ov::compilation_num_threads);
+                        } catch (const ov::Exception&) {
```

Comment:
why set here？ we already have compilation threads set here?
https://github.com/openvinotoolkit/openvino/blob/7f784509f3b791a2bb41c787c073fb9caf7f65b8/src/plugins/auto/src/auto_schedule.cpp#L320

Reply:
AUTO plugin need to use the same config of actual device as CORE level to ensure the consistent  retrieval of model hash id. So need to move the config setting here.

Reply:
and , compilation num threads is not a property that affects caching?
![image](https://github.com/openvinotoolkit/openvino/assets/32348500/0ca0a754-f013-48db-b849-9f0c85d07e73)

Reply:
You are right. this property did not affect caching hash ID. Will correct this logic at the next PR.